File: src/main/node/net/ecma/Library/Tests/Components/Interpolation/NormalInterpolator.mjs
'use strict';
var X3D = require('./x3d.mjs');
try {
	var fs = require('fs');
} catch (e) {
console.log("Problems loading fs. On browser?",e);
}
var SFString = require('./x3d.mjs');
var SFNode = require('./x3d.mjs');
var head = require('./x3d.mjs');
var MFNode = require('./x3d.mjs');
var meta = require('./x3d.mjs');
var Scene = require('./x3d.mjs');
var WorldInfo = require('./x3d.mjs');
var MetadataSet = require('./x3d.mjs');
var MetadataBoolean = require('./x3d.mjs');
var MFBool = require('./x3d.mjs');
var TimeSensor = require('./x3d.mjs');
var SFTime = require('./x3d.mjs');
var SFBool = require('./x3d.mjs');
var NormalInterpolator = require('./x3d.mjs');
var MFFloat = require('./x3d.mjs');
var MFVec3f = require('./x3d.mjs');
var Transform = require('./x3d.mjs');
var SFVec3f = require('./x3d.mjs');
var Shape = require('./x3d.mjs');
var Appearance = require('./x3d.mjs');
var PointProperties = require('./x3d.mjs');
var SFFloat = require('./x3d.mjs');
var Material = require('./x3d.mjs');
var SFColor = require('./x3d.mjs');
var PointSet = require('./x3d.mjs');
var Coordinate = require('./x3d.mjs');
var ROUTE = require('./x3d.mjs');
var X3D0 =  new X3D({

      profile : new SFString("Interchange"),
      version : new SFString("3.3"),
      head : new SFNode(
        new head({
          meta : new MFNode([
            new meta({
              name : new SFString("comment"),
              content : new SFString("World of Sunrise X3D Editor")}),

            new meta({
              name : new SFString("created"),
              content : new SFString("Mon, 13 Apr 2015 02:16:11 GMT")}),

            new meta({
              name : new SFString("creator"),
              content : new SFString("Holger Seelig")}),

            new meta({
              name : new SFString("generator"),
              content : new SFString("Sunrise X3D Editor V1.0.4, https://create3000.github.io/sunrise/")}),

            new meta({
              name : new SFString("modified"),
              content : new SFString("Fri, 28 Oct 2022 13:27:25 GMT")}),

            new meta({
              name : new SFString("identifier"),
              content : new SFString("file:///Volumes/Home/Projekte/Library/Tests/Components/Interpolation/NormalInterpolator.x3d")})])})),
      Scene : new SFNode(
        new Scene({
          children : new MFNode([
            new WorldInfo({
              title : new SFString("NormalInterpolator"),
              metadata : new MFNode([
                new MetadataSet({
                  name : new SFString("Titania"),
                  value : new SFNode(
                    new MetadataSet({
                      name : new SFString("AngleGrid"),
                      value : new SFNode(
                        new MetadataBoolean({
                          name : new SFString("enabled"),
                          value : new MFBool([true])}))})),
                  value : new SFNode(
                    new MetadataSet({
                      name : new SFString("Grid"),
                      value : new SFNode(
                        new MetadataBoolean({
                          name : new SFString("enabled"),
                          value : new MFBool([false])}))}))})])}),

            new TimeSensor({
              DEF : new SFString("_1"),
              cycleInterval : new SFTime(10),
              loop : new SFBool(true)}),

            new NormalInterpolator({
              DEF : new SFString("_2"),
              key : new MFFloat([0,0.25,0.5,0.75,1]),
              keyValue : new MFVec3f([1,0,0,0,1,0,-1,0,0,0,-1,0,0,1,0,-1,0,0,0,-1,0,1,0,0,-1,0,0,0,-1,0,1,0,0,0,1,0,0,-1,0,1,0,0,0,1,0,-1,0,0,1,0,0,0,1,0,-1,0,0,0,-1,0])}),

            new Transform({
              DEF : new SFString("PointSet_1"),
              scale : new SFVec3f([3,3,3]),
              children : new MFNode([
                new Shape({
                  appearance : new SFNode(
                    new Appearance({
                      pointProperties : new SFNode(
                        new PointProperties({
                          pointSizeScaleFactor : new SFFloat(20),
                          pointSizeMinValue : new SFFloat(20),
                          pointSizeMaxValue : new SFFloat(20)})),
                      material : new SFNode(
                        new Material({
                          emissiveColor : new SFColor([1,0,0])}))})),
                  geometry : new SFNode(
                    new PointSet({
                      coord : new SFNode(
                        new Coordinate({
                          DEF : new SFString("_3"),
                          point : new MFVec3f([-0.983304,-0.181969,0,0.181969,-0.983304,0,0.983304,0.181969,0,-0.181969,0.983304,0])}))}))})])}),

            new ROUTE({
              fromNode : new SFString("_1"),
              fromField : new SFString("fraction_changed"),
              toNode : new SFString("_2"),
              toField : new SFString("set_fraction")}),

            new ROUTE({
              fromNode : new SFString("_2"),
              fromField : new SFString("value_changed"),
              toNode : new SFString("_3"),
              toField : new SFString("set_point")})])}))});
console.log(X3D0.toXMLNode());

File: src/main/cplusplus/net/coderextreme/Library/Tests/Components/Interpolation/NormalInterpolator.cpp
#define False false
#define True true
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/IndexedFaceSetExample/pch.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/IndexedFaceSetExample/framework.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/include/glut.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/include/X3DLib.h"
int main(int argc, char ** argv) {
X3D& X3D0 =  X3D();
X3D0.setProfile(CString("Interchange"));
X3D0.setVersion(CString("3.3"));
head& head1 =  head();
meta& meta2 =  meta();
meta2.setName(CString("comment"));
meta2.setContent(CString("World of Sunrise X3D Editor"));
head1.addMeta(&meta2);

meta& meta3 =  meta();
meta3.setName(CString("created"));
meta3.setContent(CString("Mon, 13 Apr 2015 02:16:11 GMT"));
head1.addMeta(&meta3);

meta& meta4 =  meta();
meta4.setName(CString("creator"));
meta4.setContent(CString("Holger Seelig"));
head1.addMeta(&meta4);

meta& meta5 =  meta();
meta5.setName(CString("generator"));
meta5.setContent(CString("Sunrise X3D Editor V1.0.4, https://create3000.github.io/sunrise/"));
head1.addMeta(&meta5);

meta& meta6 =  meta();
meta6.setName(CString("modified"));
meta6.setContent(CString("Fri, 28 Oct 2022 13:27:25 GMT"));
head1.addMeta(&meta6);

meta& meta7 =  meta();
meta7.setName(CString("identifier"));
meta7.setContent(CString("file:///Volumes/Home/Projekte/Library/Tests/Components/Interpolation/NormalInterpolator.x3d"));
head1.addMeta(&meta7);

X3D0.setHead(&head1);

Scene& Scene8 =  Scene();
WorldInfo& WorldInfo9 =  WorldInfo();
WorldInfo9.setTitle(CString("NormalInterpolator"));
MetadataSet& MetadataSet10 =  MetadataSet();
MetadataSet10.X3DNode::setName(CString("Titania"));
MetadataSet& MetadataSet11 =  MetadataSet();
MetadataSet11.X3DNode::setName(CString("AngleGrid"));
MetadataBoolean& MetadataBoolean12 =  MetadataBoolean();
MetadataBoolean12.X3DNode::setName(CString("enabled"));
MetadataBoolean12.setValue(new boolean[]{True});
MetadataSet11.setValue((X3DNode *)&MetadataBoolean12);

MetadataSet10.setValue((X3DNode *)&MetadataSet11);

MetadataSet& MetadataSet13 =  MetadataSet();
MetadataSet13.X3DNode::setName(CString("Grid"));
MetadataBoolean& MetadataBoolean14 =  MetadataBoolean();
MetadataBoolean14.X3DNode::setName(CString("enabled"));
MetadataBoolean14.setValue(new boolean[]{False});
MetadataSet13.setValue((X3DNode *)&MetadataBoolean14);

MetadataSet10.setValue((X3DNode *)&MetadataSet13);

WorldInfo9.setMetadata(&MetadataSet10);

Scene8.addChild(&WorldInfo9);

TimeSensor& TimeSensor15 =  TimeSensor();
TimeSensor15.setDEF(CString("_1"));
TimeSensor15.setCycleInterval(10);
TimeSensor15.setLoop(True);
Scene8.addChild(&TimeSensor15);

NormalInterpolator& NormalInterpolator16 =  NormalInterpolator();
NormalInterpolator16.setDEF(CString("_2"));
NormalInterpolator16.setKey(new float[]{0,0.25,0.5,0.75,1}, 5);
NormalInterpolator16.setKeyValue(new float[]{1,0,0,0,1,0,-1,0,0,0,-1,0,0,1,0,-1,0,0,0,-1,0,1,0,0,-1,0,0,0,-1,0,1,0,0,0,1,0,0,-1,0,1,0,0,0,1,0,-1,0,0,1,0,0,0,1,0,-1,0,0,0,-1,0}, 60);
Scene8.addChild(&NormalInterpolator16);

Transform& Transform17 =  Transform();
Transform17.setDEF(CString("PointSet_1"));
Transform17.setScale(new float[]{3,3,3});
Shape& Shape18 =  Shape();
Appearance& Appearance19 =  Appearance();
PointProperties& PointProperties20 =  PointProperties();
PointProperties20.setPointSizeScaleFactor(20);
PointProperties20.setPointSizeMinValue(20);
PointProperties20.setPointSizeMaxValue(20);
Appearance19.setPointProperties(PointProperties20);

Material& Material21 =  Material();
Material21.setEmissiveColor(new float[]{1,0,0});
Appearance19.addChild(&Material21);

Shape18.addChild(&Appearance19);

PointSet& PointSet22 =  PointSet();
Coordinate& Coordinate23 =  Coordinate();
Coordinate23.setDEF(CString("_3"));
Coordinate23.setPoint(new float[]{-0.983304,-0.181969,0,0.181969,-0.983304,0,0.983304,0.181969,0,-0.181969,0.983304,0}, 12);
PointSet22.setCoord(&Coordinate23);

Shape18.setGeometry(PointSet22);

Transform17.addChild(&Shape18);

Scene8.addChild(&Transform17);

ROUTE& ROUTE24 =  ROUTE();
ROUTE24.setFromNode(CString("_1"));
ROUTE24.setFromField(CString("fraction_changed"));
ROUTE24.setToNode(CString("_2"));
ROUTE24.setToField(CString("set_fraction"));
Scene8.addChild(&ROUTE24);

ROUTE& ROUTE25 =  ROUTE();
ROUTE25.setFromNode(CString("_2"));
ROUTE25.setFromField(CString("value_changed"));
ROUTE25.setToNode(CString("_3"));
ROUTE25.setToField(CString("set_point"));
Scene8.addChild(&ROUTE25);

X3D0.setScene(&Scene8);

}
